pkg/config/manifest.go

```diff
@@ -112,6 +112,47 @@ type ParameterDefinition struct {
 	definition.Schema `yaml:",inline"`
 }
 
+func (pd *ParameterDefinition) Validate() error {
+	var result *multierror.Error
+
+	if pd.Name == "" {
+		result = multierror.Append(result, errors.New("parameter name is required"))
+	}
+
+	// Porter supports declaring a parameter of type: "file",
+	// which we will convert to the appropriate bundle.Parameter type in adapter.go
+	// Here, we copy the ParameterDefinition and make the same modification before validation
+	pdCopy := pd.DeepCopy()
+	if pdCopy.Type == "file" {
+		if pd.Destination.Path == "" {
+			result = multierror.Append(result, fmt.Errorf("no destination path supplied for parameter %s", pd.Name))
+		}
+		pdCopy.Type = "string"
+		pdCopy.ContentEncoding = "base64"
+	}
+
+	schemaValidationErrs, err := pdCopy.Schema.Validate(pdCopy)
+	if err != nil {
+		result = multierror.Append(result, errors.Wrapf(err, "encountered error while validating parameter %s", pdCopy.Name))
+	}
+	for _, schemaValidationErr := range schemaValidationErrs {
+		result = multierror.Append(result, errors.Wrapf(err, "encountered validation error(s) for parameter %s: %v", pdCopy.Name, schemaValidationErr))
+	}
+
+	return result.ErrorOrNil()
+}
+
+// DeepCopy copies a ParameterDefinition and returns the copy
+func (pd *ParameterDefinition) DeepCopy() *ParameterDefinition {
+	return &ParameterDefinition{
+		Name:        pd.Name,
+		Sensitive:   pd.Sensitive,
+		ApplyTo:     pd.ApplyTo,
```

Comment:
This is keeping the same slice and backing array for Apply to. So if you modified the copy of ApplyTo, it would modify the original.

What you can do instead to make a copy is rely on Go copy semanitcs (which copies the contents of all the fields for you automatically) and then copy the slice yourself.

https://play.golang.org/p/jVOtUyQ0rXj

Reply:
Ah, thank you!  Updated.

Reply:
STANDUP MERGE IN PROGRESS!